.github/workflows/set-metadata.yml

```diff
@@ -0,0 +1,354 @@
+      version:
+        value: ${{ jobs.set-version.outputs.version }}
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+
+  construct-payload:
+    runs-on: ubuntu-latest
+    outputs:
+      payload: ${{ steps.construct-payload.outputs.payload }}
+    steps:
+
+    - name: Construct payload
+      id: construct-payload
+      run: |
+        if [[ -n "${{ inputs.payload }}" ]]; then
+          echo "Found preconstructed JSON payload"
+          echo "payload=${{ inputs.payload }}" >> $GITHUB_OUTPUT
+          exit 0
+        fi
+
+        echo "Constructing JSON payload from raw inputs"
+        payload="{}"
+
+        # Get howso-engine-py version or run ID
+        if [[ "${{ inputs.howso-engine-py-build }}" =~ runs/([0-9]+) ]]; then
+          payload=$(echo "$payload" | jq --arg run_id "${BASH_REMATCH[1]}" '. + {"howso-engine-py": $run_id}')
+        elif [[ -n "${{ inputs.howso-engine-py-build }}" ]]; then
+          payload=$(echo "$payload" | jq --arg run_id "${{ inputs.howso-engine-py-build }}" '. + {"howso-engine-py": $run_id}')
+        fi
+
+        # Get amalgam-lang-py version or run ID
+        if [[ "${{ inputs.amalgam-lang-py-build }}" =~ runs/([0-9]+) ]]; then
+          payload=$(echo "$payload" | jq --arg run_id "${BASH_REMATCH[1]}" '. + {"amalgam-lang-py": $run_id}')
+        elif [[ -n "${{ inputs.amalgam-lang-py-build }}" ]]; then
+          payload=$(echo "$payload" | jq --arg run_id "${{ inputs.amalgam-lang-py-build }}" '. + {"amalgam-lang-py": $run_id}')
+        fi
+
+        # Get howso-validator-py version or run ID
+        if [[ "${{ inputs.howso-validator-py-build }}" =~ runs/([0-9]+) ]]; then
+          payload=$(echo "$payload" | jq --arg run_id "${BASH_REMATCH[1]}" '. + {"howso-validator-py": $run_id}')
+        elif [[ -n "${{ inputs.howso-validator-py-build }}" ]]; then
+          payload=$(echo "$payload" | jq --arg run_id "${{ inputs.howso-validator-py-build }}" '. + {"howso-validator-py": $run_id}')
+        fi
+
+        # Get howso-validator-enterprise-py version or run ID
+        if [[ "${{ inputs.howso-validator-enterprise-py-build }}" =~ runs/([0-9]+) ]]; then
+          payload=$(echo "$payload" | jq --arg run_id "${BASH_REMATCH[1]}" '. + {"howso-validator-enterprise-py": $run_id}')
+        elif [[ -n "${{ inputs.howso-validator-enterprise-py-build }}" ]]; then
+          payload=$(echo "$payload" | jq --arg run_id "${{ inputs.howso-validator-enterprise-py-build }}" '. + {"howso-validator-enterprise-py": $run_id}')
+        fi
+
+        # Get howso-synthesizer-py version or run ID
+        if [[ "${{ inputs.howso-synthesizer-py-build }}" =~ runs/([0-9]+) ]]; then
+          payload=$(echo "$payload" | jq --arg run_id "${BASH_REMATCH[1]}" '. + {"howso-synthesizer-py": $run_id}')
+        elif [[ -n "${{ inputs.howso-synthesizer-py-build }}" ]]; then
+          payload=$(echo "$payload" | jq --arg run_id "${{ inputs.howso-synthesizer-py-build }}" '. + {"howso-synthesizer-py": $run_id}')
+        fi
```

Comment:
Same as the other file, this looks like it can be collapsed into a loop with a good data dict up front.

Reply:
I was thinking just put it in the dict (or list of list) and let the pipeline do the replacement for you:

```json
{ 
 { "${{ inputs.howso-engine-py-build }}", "howso-engine-py" },
 { "${{ inputs.amalgam-lang-py-build }}", "amalgam-lang-py" }
}
```